neqo-http3/src/server_events.rs

```diff
@@ -81,6 +87,22 @@ impl ClientRequestStream {
     }
 }
 
+impl std::hash::Hash for ClientRequestStream {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        self.conn.hash(state);
```

Comment:
Yes, we need the connection too. For simplicity reasons currently we only have one server running that serves request from all test. Currently there are not so many test that needs this, so it would work without(with "execute sequential" flag for tests), but at some point we will chase the bug.